ecl/hthor/hthor.cpp

```diff
@@ -158,14 +158,16 @@ const void * CRowBuffer::next()
 }
 
 
-ILocalOrDistributedFile *resolveLFNFlat(IAgentContext &agent, const char *logicalName, const char *errorTxt, bool optional)
+ILocalOrDistributedFile *resolveLFNFlat(IAgentContext &agent, const char *logicalName, const char *errorTxt, bool optional, bool privilegedUser)
 {
     Owned<ILocalOrDistributedFile> ldFile = agent.resolveLFN(logicalName, errorTxt, optional);
     if (!ldFile)
         return nullptr;
     IDistributedFile *dFile = ldFile->queryDistributedFile();
     if (dFile && isFileKey(dFile))
         throw MakeStringException(0, "Attempting to read index as a flat file: %s", logicalName);
+    if (dFile->isRestrictedAccess() && !privilegedUser)
+        throw MakeStringException(0, "Access denied to restricted file: %s", logicalName);
```

Comment:
why is this checking here? It should already have been checked and fired exception if necessary in DFS lookup shouldn't it?

Reply:
also dFile could be null here.